Fix submission namespace output when including dashes

vknabel opened the pull request:
When a new provider or module has been submitted which included a `-` within the namespace of the Git repository, only the last portion of the user or organization was picked.

See #125 as an example where `metal-stack-cloud/terraform-provider-metal` resulted in a errornous repository url of `https://github.com/cloud/terraform-provider-metal`.

Yantrio commented on lambda.tf:
```diff
@@ -58,30 +59,30 @@ resource "aws_lambda_function" "api_function" {
-      GITHUB_TOKEN_SECRET_ASM_NAME             = aws_secretsmanager_secret.github_api_token.name
-      PROVIDER_NAMESPACE_REDIRECTS             = jsonencode(var.provider_namespace_redirects)
-      PROVIDER_VERSIONS_TABLE_NAME             = aws_dynamodb_table.provider_versions.name
+      GITHUB_TOKEN_SECRET_ASM_NAME = aws_secretsmanager_secret.github_api_token.name
```
can you revert this formatting change please

eranelbaz replied:
dfc4e976(rollback formatting)

Yantrio commented on lambda.tf:
```diff
@@ -58,30 +59,30 @@ resource "aws_lambda_function" "api_function" {
   handler       = "populate-provider-versions"
   memory_size   = 128
-  timeout       = 10 * 60
+  timeout       = 60
```
this timeout should be 10 minutes, not 60 seconds

eranelbaz replied:
dfc4e976(rollback formatting)

Yantrio commented on lambda.tf:
```diff
@@ -58,30 +59,30 @@ resource "aws_lambda_function" "api_function" {
 // ensure we have provisioned concurrency for the lambda function
 resource "aws_lambda_provisioned_concurrency_config" "api_function" {
-  function_name                     = aws_lambda_function.api_function.function_name
+  function_name = aws_lambda_function.api_function.function_name
```
revert this formatting change please.

eranelbaz replied:
dfc4e976(rollback formatting)

Yantrio reviewed (changes requested):
Overall this looks good, just a couple nitpicks.

eranelbaz commented:
@Yantrio rollback those formatting issues + timeouts

vknabel committed:
Fix submission namespace output when including dashes

When a new provider or module has been submitted which included a `-` within the namespace of the Git repository, only the last portion of the user or organization was picked.

See #125 as an example where `metal-stack-cloud/terraform-provider-metal` resulted in a errornous repository url of `https://github.com/cloud/terraform-provider-metal`.

Signed-off-by: Valentin Knabel <dev@vknabel.com>

ghost commented:
@vknabel thank you for reporting this, it looks to me as if that regexp you changed could do with a little more attention from our side. It may need more than this change, but we'll try to sort it out as quickly as possible since this seems to be a blocker.

vknabel commented on src/cmd/add-module/main.go:
```diff
@@ -45,7 +45,7 @@ func main() {
 	err = func() error {
 		// Lower case input
-		re := regexp.MustCompile("(?P<Namespace>[a-zA-Z0-9]+)/terraform-(?P<Target>[a-zA-Z0-9]*)-(?P<Name>[a-zA-Z0-9-]*)")
+		re := regexp.MustCompile("(?P<Namespace>[a-zA-Z0-9-]+)/terraform-(?P<Target>[a-zA-Z0-9]*)-(?P<Name>[a-zA-Z0-9-]*)")
```
Thanks for the fast feedback!

In case we use `^` and `$` to match the whole line, we also need to iterate over the comment line by line. (`^.+$` matches the whole multiline string)

e.g.

```go
re := regexp.MustCompile("^(?P<Namespace>[a-zA-Z0-9-]+)/terraform-(?P<Target>[a-zA-Z0-9]*)-(?P<Name>[a-zA-Z0-9-]*)$")
repoScanner := bufio.NewScanner(strings.NewReader(*repository))
var match []string
for len(match) == 0 && repoScanner.Scan() {
   match = re.FindStringSubmatch(repoScanner.Text())
}
```


Alternatives for matching the namespace taken from this repository:
* `[a-zA-Z0-9-]+` current _(too specific?)_
* `.*` in `internal/v1api/providers.go` `func ArchivedOverrides` _(too broad?)_
* `[^/]+` in `internal/provider/list.go` `providerDirectoryRegex` _(my favorite)_


Happy to implement your decision.

cam72cam replied:
This should work fine.  The match below requires the full string to match.
